store/sqlstore/post_store.go

```diff
@@ -600,54 +635,81 @@ func (s *SqlPostStore) getPostsAround(channelId string, postId string, limit int
 	}
 
 	var posts, parents []*model.Post
-	_, err := s.GetReplica().Select(&posts,
-		`SELECT
-			*
-		FROM
-			Posts
-		WHERE
-			CreateAt `+direction+` (SELECT CreateAt FROM Posts WHERE Id = :PostId)
-				AND ChannelId = :ChannelId
-				AND DeleteAt = 0
-		ORDER BY CreateAt `+sort+`
-		LIMIT :Limit
-		OFFSET :Offset`,
-		map[string]interface{}{"ChannelId": channelId, "PostId": postId, "Limit": limit, "Offset": offset})
-	if err != nil {
-		return nil, model.NewAppError("SqlPostStore.GetPostContext", "store.sql_post.get_posts_around.get.app_error", nil, "channelId="+channelId+err.Error(), http.StatusInternalServerError)
-	}
-
-	if len(posts) > 0 {
-		rootIds := []string{}
-		for _, post := range posts {
-			rootIds = append(rootIds, post.Id)
-			if post.RootId != "" {
-				rootIds = append(rootIds, post.RootId)
-			}
+	if skipRootFetch {
+		_, err := s.GetReplica().Select(&posts,
+			`SELECT
+			p.*,
+			(
+				SELECT
+					count(Posts.Id)
+				FROM
+					Posts
+				WHERE
+					(p.RootId != '' AND Posts.RootId = p.RootId) OR (p.RootId = '' AND Posts.RootId = p.Id)) AS ResponseCount
```

Comment:
The indentation here is confusing -- can we wrap the `) AS ResponseCount` down to a single line and align with the opening `)`?

Also, I don't follow the conditions here -- why do we need to generate a `ThreadReplyCount` when `p.RootId != ''`? Presumably we only need count the replies to the root post?

Reply:
@lieut-data - we send ReplyCount for each post. You don't always have 'root post' in your store or in your response, unless I am missing something...

Reply:
It's find to send it, but there's no reason to compute it when `p.RootId != ''` -- such a post /is/ a reply, and thus would always have a `ReplyCount` of `0`.

Reply:
@lieut-data I think I understand our misunderstanding here. The ReplyCount is the total length of the thread the post participates in. It's not a number of replies to the post

Reply:
@reflog, the total length of the thread is constant for all posts in that thread, and it's equivalent to the number of replies to the /root/ post -- right?

My point is that we need only calculate this for the root post, never for any reply to that thread.

Reply:
> You don't always have 'root post' in your store or in your response, unless I am missing something...

Ah, but this is the key nuance that I missed in your response: you're right that we will (no longer) have the root post in the store to rely upon. This is unfortunate, and raises the same issue in the webapp discussion: how do we keep this up-to-date for other posts that have already been fetched?

I'm wondering if we should tweak the model here to go from "post + thread" to "post + root" instead of just "post", so that we guarantee we always have the root post, and then always have the count for that root post. Every time we fetch a new reply post, we'd refresh the root post and thus the corresponding reply count, to be rendered everywhere the same.

Reply:
I gave this some more thought, and I think we /have/ to return the root posts, otherwise we can't guarantee the ability to quote the "replied to ..." in the user interface. Or are we changing the UX here?

Reply:
@lieut-data I completely agree. This solution will be much more robust. I'll add the root post to the returned values. 💪